File: Latihan A/cpp/Pc.cpp
// Library dan kelas
#include <iostream>
#include <string>
using namespace std;
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"
// deklarasi kelas
class Pc{
    // variabel dan objek private
    private:
        Processor Pcproc;
        Disk Pcdisk;
        Ram Pcram;
        int totalprice;
    // variabel dan method publik
    public:
        // konstruktor
        Pc(){

        }
        Pc(Processor Pcproc, Disk Pcdisk, Ram Pcram){
            this->Pcproc = Pcproc;
            this->Pcdisk = Pcdisk;
            this->Pcram = Pcram;
            this->totalprice = Pcproc.getprice() + Pcdisk.getprice() + Pcram.getprice();
        }
        // get dan set
        void setPcproc(Processor Pcproc){
            this->Pcproc = Pcproc;
        }
        Processor getPcproc(){
            return this->Pcproc;
        }
        void setPcram(Ram Pcram){
            this->Pcram = Pcram;
        }
        Ram getPcram(){
            return this->Pcram;
        }
        void setPcdisk(Disk Pcdisk){
            this->Pcdisk = Pcdisk;
        }
        Disk getPcdisk(){
            return this->Pcdisk;
        }
        // method tampilkan
        void printpc(){
            cout << "Spesifikasi PC: " << endl;
            cout << "Processor: " << this->Pcproc.getname()  << endl;
            cout << "Processor Price: " << this->Pcproc.getprice()  << endl;
            cout << "Disk Type: " << this->Pcdisk.gettype()  << endl;
            cout << "Disk Capacity: " << this->Pcdisk.getcapacity()  << endl;
            cout << "Disk Price: " << this->Pcdisk.getprice()  << endl;
            cout << "Ram Capacity: " << this->Pcram.getcapacity()  << endl;
            cout << "Ram Price: " << this->Pcram.getprice()  << endl;
            cout << "Total Price: " << this->totalprice  << endl;
        }
        // destruktor
        ~Pc(){}
};

File: Latihan B/php/Pc.php
<?php
// deklarasi kelas
class Pc{
	//variabel lokal
	private $Processor;
	private $Disk;
    private $Ram;
    private $totalprice;
	//konstruktor
	public function __construct($Processor, $Disk, $Ram, $totalprice){
		$this->Processor = $Processor;
        $this->Disk = $Disk;
        $this->Ram = $Ram;
        $this->totalprice = $totalprice;
	}
	//get dan set setiap variabel
	public function settotalprice($totalprice){
		$this->totalprice = $totalprice;
	}
	public function gettotalprice(){
		return $this->totalprice;
	}
	public function setProcessor($Processor){
		$this->Processor = $Processor;
	}
	public function getProcessor(){
		return $this->Processor;
	}
    public function setRam($Ram){
		$this->Ram = $Ram;
	}
	public function getRam(){
		return $this->Ram;
	}
    public function setDisk($Disk){
		$this->Disk = $Disk;
	}
	public function getDisk(){
		return $this->Disk;
	}
	// fungsi tampilkan
    public function print(){
        echo "Processor Name : ". $this->Processor->getname() ."<br>";
        echo "Processor Price : ". $this->Processor->getprice() ."<br>";
        echo "Disk Type : ". $this->Disk->gettype() ."<br>";
        echo "Disk Capacity : ". $this->Disk->getcapacity() ."<br>";
        echo "Disk Price : ". $this->Disk->getprice() ."<br>";
        echo "Ram Capacity : ". $this->Ram->getcapacity() ."<br>";
        echo "Ram Price : ". $this->Ram->getprice() ."<br>";
        echo "Total Price : ". $this->totalprice ."<br>";
    }
	// destruktor
	public function __destruct(){
	}
}
?>